src/devices/RGBDFromRosTopic/RGBDFromRosTopic.cpp

```diff
@@ -0,0 +1,380 @@
+    return m_rgb_input_processor->getIntrinsicParam(intrinsic);
+}
+
+int  RGBDFromRosTopic::getDepthHeight()
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        return  0;
+    }
+    return (int)m_depth_input_processor->getHeight();
+}
+
+int  RGBDFromRosTopic::getDepthWidth()
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        return  0;
+    }
+    return (int)m_depth_input_processor->getWidth();
+}
+
+bool RGBDFromRosTopic::getDepthFOV(double& horizontalFov, double& verticalFov)
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        horizontalFov = 0;
+        verticalFov = 0;
+        return true;
+    }
+    return m_depth_input_processor->getFOV(horizontalFov, verticalFov);
+}
+
+bool RGBDFromRosTopic::getDepthIntrinsicParam(Property& intrinsic)
+{
+    if (m_depth_input_processor == nullptr)
+    {
+        intrinsic.clear();
+        return true;
+    }
+    return m_depth_input_processor->getIntrinsicParam(intrinsic);
+}
+
+double RGBDFromRosTopic::getDepthAccuracy()
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthAccuracy not supported";
+    return 0;
+}
+
+bool RGBDFromRosTopic::getDepthClipPlanes(double& nearPlane, double& farPlane)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthClipPlanes not supported";
+    return false;
+}
+
+bool RGBDFromRosTopic::setDepthClipPlanes(double nearPlane, double farPlane)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "setDepthClipPlanes not supported";
+    return false;
+}
+
+bool RGBDFromRosTopic::getDepthMirroring(bool& mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getDepthMirroring not supported";
+    return false;
+}
+
+bool RGBDFromRosTopic::setDepthMirroring(bool mirror)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "setDepthMirroring not supported";
+    return false;
+}
+
+bool RGBDFromRosTopic::getExtrinsicParam(Matrix& extrinsic)
+{
+    yCWarning(RGBD_ROS_TOPIC) << "getExtrinsicParam not supported";
+    return  false;
+};
+
+bool RGBDFromRosTopic::getRgbImage(FlexImage& rgbImage, Stamp* timeStamp)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor!=nullptr)
+        bool rgb_ok = m_rgb_input_processor->getLastRGBData(rgbImage, *timeStamp);
+    return true;
+}
+
+bool RGBDFromRosTopic::getDepthImage(ImageOf<PixelFloat>& depthImage, Stamp* timeStamp)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_depth_input_processor != nullptr)
+        bool depth_ok = m_depth_input_processor->getLastDepthData(depthImage, *timeStamp);
+    return true;
+}
+
+bool RGBDFromRosTopic::getImages(FlexImage& colorFrame, ImageOf<PixelFloat>& depthFrame, Stamp* colorStamp, Stamp* depthStamp)
+{
+    bool rgb_ok = false;
+    bool depth_ok = false;
+    std::lock_guard<std::mutex> guard(m_mutex);
+    if (m_rgb_input_processor != nullptr)
+        rgb_ok = m_rgb_input_processor->getLastRGBData(colorFrame, *colorStamp);
+    if (m_depth_input_processor != nullptr)
+        depth_ok = m_depth_input_processor->getLastDepthData(depthFrame, *depthStamp);
```

Comment:
Codacy found an issue: [Variable 'depth_ok' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6068551)